### ISSUE TITLE
Add metatensor-atomistic example with neighborlist

### ISSUE THREAD

**Issue by PicoCentauri** [lab-cosmo/metatensor#701]
We have a good example explaining how to create a model and export one. However, we don't show how to request and use a neighborlist (using `requested_neighbor_lists()`) in any of our example. The use is explained in the docs but I think an example with a neighborlist should help.

The example basically boils down to a code block like

```python
class Model
    def __init__(self):
        self.nl = NeighborListOptions(...)

    def requested_neighbor_lists(self):
        return [self.nl]

    def forward(self, systems, ...):
        neighbors = systems[0].get_neighbor_list(self.nl)
```

and maybe running this code somehow.

**Comment by Luthaf** [comment]
Such an example could re-use the code from https://github.com/Luthaf/metatensor-lj-test/blob/main/src/metatensor_lj_test/pure.py, which implements a basic Lennard-Jones model.

**Comment by PicoCentauri** [comment]
Yes, I was about to write the same thing xD. I can try to simplify the code from your repo and move the important things to an example. I suggest changing the order of the examples to 

1. export-atomistic-model
2. running-ase-md.py
3. atomistic-model-with-nl
4. profiling
